include/private/libevent/dns.hpp

```diff
@@ -335,6 +335,22 @@ void query(QueryClass dns_class, QueryType dns_type, std::string name,
            Callback<Error, Var<Message>> cb, Settings settings,
            Var<Reactor> reactor, Var<Logger> logger) {
 
+    /*
+     * When running OONI tests, we're interested to know not only the IPs
+     * associated with a specific name, but also to the CNAME.
+     *
+     * This is not yet implemented in the libevent engine.
+     */
+    ErrorOr<bool> also_cname = settings.get("dns/resolve_also_cname", false);
+    if (!also_cname) {
+        cb(also_cname.as_error(), nullptr);
+        return;
+    }
+    if (*also_cname == true) {
+        cb(mk::NotImplementedError(), nullptr);
```

Comment:
ha, so libevent doesn't support this?

Reply:
It is probably possible to do that with some hacking. I recall there is a way to convince it doing so, but decided not to bother implementing it, since we're using the `system` resolver for all cases except when we need to force a specific DNS server. Since the plan is to switch over to `c-ares` in the future, I figured it was more efficient not to spend time on hammering this further.